### PR TITLE
Add support for `PathLike` objects

### PR DESCRIPTION
**Pull request by phill-85**
Split file handling in to separate function for easier testing

### CODE REVIEW

**Comment by sbraz** [comment]
Thanks for the PR, can I change it a bit before merging it?

**Comment by phill-85** [comment]
Sure, go ahead, though I'm curious as to what you want to change?

**Comment by sbraz** [comment]
Mostly the grammar and some spacing. I'll show you before merging.

File: pymediainfo/__init__.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def parse(cls, filename, library_file=None, cover_data=False,
         >>> pymediainfo.MediaInfo.parse("tests/data/sample.mkv",
         ...     library_file="/path/to/libmediainfo.dylib")
 
-        :param filename: path to the media file which will be analyzed.
+        :param str | Path | os.PathLike filename: path to the media file which will be analyzed.
```
Comment (sbraz) [inline pymediainfo/__init__.py]: Sphinx doesn't like those pipes.

File: tests/test_pymediainfo.py [hunk 2]
```diff
@@ -112,6 +117,33 @@ def test_parse_non_existent_path_pathlib(self):
         path = self.pathlib.Path(data_dir) / "this file does not exist"
         self.assertRaises(FileNotFoundError, MediaInfo.parse, path)
 
+
+class MediaInfoTestFilenameTypes(unittest.TestCase):
+    def test_convert_filename_str(self):
```
Comment (sbraz) [inline tests/test_pymediainfo.py]: You used the same name for all three methods so only the last one is actually run.

File: tests/test_pymediainfo.py [hunk 3]
```diff
@@ -112,6 +117,33 @@ def test_parse_non_existent_path_pathlib(self):
...
+        path = os.path.join(data_dir, "test.txt")
+        filename, url = convert_filename(path)
+        self.assertEqual(filename, path)
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 4),
+        reason="Were testing the pathlib route of execution, but pathlib only shipped with py3.4+")
```
Comment (sbraz) [inline tests/test_pymediainfo.py]: I'm gonna `importorskip` pathlib as I did for the other pathlib test, I might even merge those test classes. Pathlib can be available for older versions via backports.

File: pymediainfo/__init__.py (hunk 1 above)
Comment (sbraz) [reply pymediainfo/__init__.py]: The type was defined below too.

**Comment by sbraz** [comment]
Hi @phill-85, I'm about to merge this, however, can you please explain what advantage over `pathlib.PurePath` this has? Is it just more standard and future-proof?

**Pull request closed without merging**